turbomoleio/core/utils.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# The turbomoleio package, a python interface to Turbomole
+# for preparing inputs, parsing outputs and other related tools.
+#
+# Copyright (C) 2018-2021 BASF SE, Matgenix SRL.
+#
+# This file is part of turbomoleio.
+#
+# Turbomoleio is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Turbomoleio is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with turbomoleio (see ~turbomoleio/COPYING). If not,
+# see <https://www.gnu.org/licenses/>.
+
+"""
+Module containing miscellaneous utilities
+"""
+
+from monty.tempfile import ScratchDir
+import re
+import pexpect.popen_spawn
+import signal
+
+
+def define_quit():
+    """Runs define but directly quits by sending "qq" interactively.
+    """
+    with open('define.log', 'wb') as logf:
+        spawn = pexpect.popen_spawn.PopenSpawn('define', timeout=2, logfile=logf)
+        try:
+            spawn.expect(["D E F I N E"])
+            spawn.sendline('qq')
+        finally:
+            if spawn.proc.poll() is None:
+                spawn.kill(sig=signal.SIGKILL)
+
+
+def get_tm_version():
+    """Get the turbomole version currently in use.
+
+    This basically runs a define and exits at the very beginning and extracts
+    the version from the header of the output.
+    """
+    with ScratchDir('.'):
+        define_quit()
+        with open('define.log', 'r') as f:
+            data = f.read()
+    r = r'TURBOMOLE.*V([\d\.]+\d)\s+.*'
+    match_version = re.search(r, data)
```

Comment:
Here it should be possible to use directly `TurbomoleData.from_file()`

Reply:
I actually tried and it did not work for all the define.log headers. Indeed the regex of the parser indicates that there should always be the build between parenthesis and that it should be composed of digits.

Sometimes there is no build, e.g. : 
 define (localhost.localdomain): TURBOMOLE rev. V7.5.1 compiled 23 Dec 2020 at 11:32:14
Sometimes the build is made of letters:
 define (localhost.localdomain) : TURBOMOLE rev. V7.4.1 (b987a8a8) compiled 15 Oct 2019 at 12:03:25

But you are right, we should do it with the TurbomoleData. I will modify the parsing for the TurbomoleData as well then.

Reply:
This should be fixed now in the parser and TurbomoleData is used to get the version.